crates/module-system/module-implementations/sov-evm/tests/integration/transactions.rs

```diff
@@ -131,3 +131,69 @@ fn test_account_nonce() {
         }),
     });
 }
+
+// Check that if the same account deploys two contracts, each deployment results in a unique contract address
+#[test]
+fn test_deploy_many_contracts() {
+    let (mut runner, account, _) = setup();
+    let contract = SimpleStorageContract::default();
+    let contract_addr_1 = account.address().create(0);
+
+    // Deploy a contract and execute two transactions that update its storage.
+    {
+        let create_contract_tx_1 = create_deploy_tx(0, &contract, &account);
+        let set_value_tx_1 = create_set_arg_tx(5, 1, &contract, contract_addr_1, &account);
+        let set_value_tx_12 = create_set_arg_tx(7, 2, &contract, contract_addr_1, &account);
+
+        let evm = Evm::<S>::default();
+        runner.execute_batch(BatchTestCase {
+            input: vec![create_contract_tx_1, set_value_tx_1, set_value_tx_12].into(),
+            assert: Box::new(move |_ctx, state| {
+                let storage_value_1 = evm
+                    .get_storage(&contract_addr_1, &U256::ZERO, state)
+                    .unwrap()
+                    .unwrap();
+
+                assert_eq!(U256::from(7), storage_value_1);
+            }),
+        });
+    }
+
+    // Deploy another contract and execut single transaction that updates its storage.
+    {
+        let contract_addr_2 = account.address().create(3);
+        let create_contract_tx_2 = create_deploy_tx(3, &contract, &account);
+        let set_value_tx_2 = create_set_arg_tx(99, 4, &contract, contract_addr_2, &account);
```

Comment:
Let's try to use simplest values. If we need two values - let's use 1 and 2. Then later the reader sees that those are obviously dummy values that don't have any additional semantics.
With 99 the reader might think - why did they choose this particular number?